modules/apps/transfer/types/coin.go

```diff
@@ -40,9 +40,15 @@ func GetPrefixedDenom(portID, channelID, baseDenom string) string {
 	return fmt.Sprintf("%s/%s/%s", portID, channelID, baseDenom)
 }
 
+// GetIBCDenom generates the full IBC denomination string based on
+// the provided portID, channelID, and baseDenom.
+func GetIBCDenom(portID, channelID, baseDenom string) string {
```

Comment:
true to the purpose of this PR, I'm seeing a couple places in our code where this might be used. I grepped for usages of `ParseDenomTrace`, there's some noise there but not too much (I think its also used in tests which also means indirect testing!).

Reply:
I don't want this function to be used a lot in the internal workings of the modules. This is meant to be useful to external developers who want to import our types.

I think internally, denom traces make sense, and `ParseDenomTrace` may have other reasons to exist (although you did mention there isn't that much noise). I think it being used in `GetTransferCoin` will indirectly test it.

I can refactor further like you suggested if others agree.